app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp

```diff
@@ -433,22 +434,25 @@ struct DeviceDelegateOculusVR::State {
         trackpadTouched = (controllerState.inputState.Touches & ovrTouch_Joystick) != 0;
         trackpadX = controllerState.inputState.Joystick.x;
         trackpadY = controllerState.inputState.Joystick.y;
-        axes[0] = trackpadX;
-        axes[1] = -trackpadY; // We did y axis intentionally inverted in FF desktop as well.
+        const int32_t kNumAxes = 4;
+        float axes[kNumAxes];
+        axes[(uint32_t)ImmersiveAxis::TouchpadX] = axes[(uint32_t)ImmersiveAxis::TouchpadY] = 0.0f;
+        axes[(uint32_t)ImmersiveAxis::ThumbstickX] = trackpadX;
+        axes[(uint32_t)ImmersiveAxis::TouchpadY] = -trackpadY; // We did y axis intentionally inverted in FF desktop as well.
         controller->SetScrolledDelta(controllerState.index, -trackpadX, trackpadY);
 
         const bool gripPressed = (controllerState.inputState.Buttons & ovrButton_GripTrigger) != 0;
-        controller->SetButtonState(controllerState.index, ControllerDelegate::BUTTON_OTHERS, 2, gripPressed, gripPressed,
-                                   controllerState.inputState.GripTrigger);
+        controller->SetButtonState(controllerState.index, ControllerDelegate::BUTTON_OTHERS, (int32_t)ImmersiveButton::Squeeze,
+                gripPressed, gripPressed, controllerState.inputState.GripTrigger);
         if (controllerState.hand == ElbowModel::HandEnum::Left) {
           const bool xPressed = (controllerState.inputState.Buttons & ovrButton_X) != 0;
           const bool xTouched = (controllerState.inputState.Touches & ovrTouch_X) != 0;
           const bool yPressed = (controllerState.inputState.Buttons & ovrButton_Y) != 0;
           const bool yTouched = (controllerState.inputState.Touches & ovrTouch_Y) != 0;
           const bool menuPressed = (controllerState.inputState.Buttons & ovrButton_Enter) != 0;
 
-          controller->SetButtonState(controllerState.index, ControllerDelegate::BUTTON_X, 3, xPressed, xTouched);
-          controller->SetButtonState(controllerState.index, ControllerDelegate::BUTTON_Y, 4, yPressed, yTouched);
+          controller->SetButtonState(controllerState.index, ControllerDelegate::BUTTON_X, 4, xPressed, xTouched);
+          controller->SetButtonState(controllerState.index, ControllerDelegate::BUTTON_Y, 5, yPressed, yTouched);
```

Comment:
If we are going to have a `const` for some indices we should have them for all of them an not mix them with hard coded values.

Reply:
Good thought!